cli/cmd/install.go

```diff
@@ -717,6 +744,31 @@ func exitIfClusterExists() {
 	os.Exit(1)
 }
 
+// exitIfNamespaceDoesNotExist checks the kubernetes API to determine if the
+// control-plane namespace exists, and returns an error if it does not.
+//
+// This is useful when running `linkerd install control-plane`, where the
+// namespace must exist, but `linkerd-config` should not.
+func exitIfNamespaceDoesNotExist() {
+	hc := newHealthChecker(
+		[]healthcheck.CategoryID{healthcheck.KubernetesAPIChecks},
+		time.Time{},
+	)
```

Comment:
TIOLI: we might gain some consistency by reusing this same minimalistic healthchecker in `exitIfClusterExists`. That might need some new `hc.CheckConfigMap` method to replace the direct call we're making in `existIfClusterExists`.

Reply:
Agree, I looked at refactoring this a bit further, then backed it out in the interest of getting this PR through. I captured your comment, and related requirements for healthcheck and client creation via #2735.